### PR TITLE
feat: support dynamic CKB fee rate

### PR DESCRIPTION
**Pull request by chaoticlonghair**

### CODE REVIEW

File: src/cli/mod.rs [hunk 1]
```diff
@@ -92,7 +93,38 @@ pub struct CkbRoArgs {
...
+    ///
+    /// For security, there is a hard limit of the dynamic ckb fee rate: `1_0000_0000`.
+    ///
+    /// Ref: <https://github.com/nervosnetwork/ckb/tree/v0.114.0/rpc#method-get_fee_rate_statistics>
+    #[arg(
+        group = "dynamic-fee-rate",
+        conflicts_with = "fixed-fee-rate",
+        long = "enable-dynamic-ckb-fee-rate"
+    )]
+    enable_dynamic: bool,
```
Comment (Flouse) [inline src/cli/mod.rs:127]: LGTM.
It's flexible since a BTC-SPV-service maintainer can choose to enable it or not.

File: src/cli/mod.rs [hunk 2]
```diff
@@ -160,6 +192,36 @@ impl CkbArgs {
...
     }
+
+    pub fn fee_rate(&self) -> Result<u64> {
+        let value = if let Some(fixed) = self.fee_rate.value() {
+            log::info!("CKB fee rate: {} (fixed)", FeeRate(fixed));
+            fixed
+        } else {
+            let dynamic = self.client().dynamic_fee_rate()?;
+            log::info!("CKB fee rate: {} (dynamic)", FeeRate(dynamic));
+            if dynamic > 1_0000_0000 {
```
Comment (Flouse) [inline src/cli/mod.rs:203]: In the beginning of 2024, there was an instance where the fee rate exceeded 1_0000_0000.
According to my recollection, the fee rate reached as high as 20_0000_0000+.

Comment (Flouse) [reply src/cli/mod.rs:203]: image

**Review by Flouse — APPROVED** [review]

File: src/cli/mod.rs [hunk 3]
```diff
@@ -160,6 +192,36 @@ impl CkbArgs {
...
+                let msg = format!(
+                    "dynamic CKB fee rate {} is too large, it seems unreasonable;\
+                    if you can confirm it, please set it as fixed fee rate",
+                    FeeRate(dynamic)
+                );
+                return Err(Error::other(msg));
```
Comment (EthanYuan) [inline src/cli/mod.rs:209]: If the dynamic fee rate exceeds the threshold 1_0000_0000, the service crashes.

Comment (Flouse) [reply src/cli/mod.rs:209]: What making MAX_DYNAMIC_FEE_RATE configurable and defaulting to 1_0000_0000?

**Review by EthanYuan — APPROVED** [review]

**Pull request merged**